src/main/java/slimeknights/tconstruct/tables/inventory/table/toolstation/ToolStationInventoryWrapper.java

```diff
@@ -0,0 +1,61 @@
+package slimeknights.tconstruct.tables.inventory.table.toolstation;
+
+import com.google.common.collect.Lists;
+import lombok.Getter;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.library.recipe.inventory.ISingleItemInventory;
+import slimeknights.tconstruct.library.recipe.inventory.InventoryIterator;
+import slimeknights.tconstruct.library.recipe.material.MaterialRecipe;
+import slimeknights.tconstruct.library.recipe.toolstation.IToolStationInventory;
+import slimeknights.tconstruct.tables.tileentity.table.ToolStationTileEntity;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ToolStationInventoryWrapper implements IToolStationInventory {
+
+  private final ToolStationTileEntity toolStation;
```

Comment:
I think we want a list of itemstacks here. If the list is null, we calculate it when calling `getInputStacks`. Whenever we add an item to slots 1-5, call a method here (not part of the inventory) to null that field.